### DISCUSSION TITLE
How to make the geological model more natural along a reverse fault

### DISCUSSION THREAD

**Discussion by Changwan-planet** [gempy-project/gempy#874] [Q&A] [answered]
Hello,

Thank you for opening this nice program. 
I am making a geological model by using information obtained from multiple trench sections across a fault.

This question is related to the previous post: https://github.com/cgre-aachen/gempy/discussions/814.
In the post, @AlexanderJuestel  Ask me "is it necessary to distinguish them?" 
Then, I didn't follow. 
My bad... Now, I do not distinguish the two basements on both side of the fault.

The model has an interface between soft sediment and basement.
The geological structure has a reverse fault, and the interface also reflects the reverse deformation. The vertical displacement is changing along the fault.

I think I am almost done with making model, but it looks strange in the part of the reverse fault. 

<img width="1280" alt="Geomodel_Miho" src="https://github.com/cgre-aachen/gempy/assets/56919733/24166671-4f3f-49e8-921d-9bf98324f063">

<img width="1280" alt="Geomodel_Miho2" src="https://github.com/cgre-aachen/gempy/assets/56919733/19cdf948-f7b0-4557-8e1d-d7cdfeb0c033">

I uploaded two image of my recent geological model. The red line, which I drew, is the fault in both images. I think the model along the strike of the fault looks unnatural, specially inside of the green circle in the second image. 

I think I am missing something. What is the problem? and how to look better? Which factor do I have to change?

I attached Points, Orientation, and Jupyter notebook (need to unzip the file for use). .

[Miho_Points.csv](https://github.com/cgre-aachen/gempy/files/14980594/Miho_Points.csv)
[Miho_Orientations.csv](https://github.com/cgre-aachen/gempy/files/14980600/Miho_Orientations.csv)
[Miho_3D_modeling_integrate.zip](https://github.com/cgre-aachen/gempy/files/14980913/Miho_3D_modeling_integrate.zip)

If you need more info, feel free to ask me. 
Thanks

**Comment by javoha** [comment]
Hi,
can you explain why you are not adding the fault as a fault to the model? I do believe that would improve your result.
Cheers

**Reply by Changwan-planet** [reply]
@javoha Many thanks for your comment.
Nice suggestion. 

I added the fault in my model. 
But I got this error 
```
ValueError: Inconsistent shapes in StacksStructure.
```

Can you review my data and jupyter notebook?

I attached the files : Points (with fault), Orientations, and Jupyter notebook.

What is my mistake?

[Miho_Question2.zip](https://github.com/gempy-project/gempy/files/15028041/Miho_Question2.zip)

**Reply by javoha** [reply]
Hi, 
looking at your data it seems that all your orientations are still labelled "SOFT", while the ones corresponding to the fault should be "FAULT".

**Reply by Changwan-planet** [reply]
![image](https://github.com/gempy-project/gempy/assets/56919733/6e460478-c2d4-41e5-9a33-129f01aa18e5)

@javoha Thank you so much! 
I improved my model from your help!!
**My next question is, How to change the color of the basement?**
I was able to change the colors of other elements. 
Most examples in the document show the color of the basement is mustard like my model.

I uploaded again my code and dataset.
[Question_Gempy2.zip](https://github.com/gempy-project/gempy/files/15079739/Question_Gempy2.zip)

**Reply by Changwan-planet** [reply]
*Another question @javoha 
In the [example](https://docs.gempy.org/examples/real/Claudius.html#sphx-glr-examples-real-claudius-py), the points of the fault are mainly located under the element 250 (color magenta). But the final model shows that the fault model is extended to the element 0 (color blue). 
If I would like the fault model to generate below the element 250, Which factor do I have to control?

Based on your comment, I hope to apply the knowledge in my model to limit the size of the fault model.

**Reply by javoha** [reply, marked as answer]
Hi,
I think not being able to change the color of the basement is a bug at the moment - I had the same issue. It will hopefully be fixed in the future (or you can of course contribute yourself and fix it for everyone).
Regarding the second question: You probably need to read a little more into the underlying structure of gempy. Which units are offset by a fault depends on timing/age relation. Still faults will always be displayed as continious over the model, as a visualization artefact, even if they do not affect all units.

**Reply by Changwan-planet** [reply]
@javoha 
Thank you, Thank you so much. 
Your answer is greatly helpful for me with quick response.
I will read more about gempy!
